File: WorkoutTrackerAPI/Data/Models/UserModels/WorkoutRecord.cs
using WorkoutTrackerAPI.Data.Models.UserModels;

namespace WorkoutTrackerAPI.Data.Models.WorkoutModels;

public class WorkoutRecord : IDbModel
{
    public long Id { get; set; }
    public TimeSpan Time { get; set; }
    public DateTime Date { get; set; }

    public long WorkoutId { get; set; }
    public Workout? Workout { get; set; }

    public string UserId { get; set; } = null!;
    public User? User { get; set; }

    public IEnumerable<ExerciseRecordGroup> ExerciseRecordGroups { get; set; } = null!;
}

File: WorkoutTracker/src/app/workout-records/workout-record.ts
import { DbModel } from "src/app/shared/models/db-model";
import { TimeSpan } from "src/app/shared/models/time-span";
import { ModelWeight } from "../shared/models/model-weight";
import { ExerciseRecordGroup } from "../shared/models/exercise-record-group";
import { Exercise } from "../exercises/models/exercise";

export interface WorkoutRecord extends DbModel {
    time: TimeSpan;
    date: Date;
    weight: ModelWeight;

    workoutId: number;
    workoutName: string;

    exercises: Exercise[];
    exerciseRecordGroups: ExerciseRecordGroup[];
}